core-js/src/test/javascript/cdf/components/UnmanagedComponent-spec.js

```diff
@@ -217,12 +221,97 @@ define([
 
       dashboard.update(uqueryWithParams);
     });
+   
+   /**
+    * ## Unmanaged Component # should return one of the known query definition properties
+    *
+    */
+    it("should return one of the known query definition properties", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      var querydefinition = { dataSource: "otherDatasource"};
+      uquery.queryDefinition = querydefinition ;
+      uquery.trafficDefinition = undefined;
+      uquery.chartDefinition = undefined;
+      expect(uquery.getQueryDefinition()).toBe(querydefinition)
+      uquery = renewUquery();
+      uquery.chartDefinition = querydefinition;
+      uquery.queryDefinition = undefined;
+      uquery.trafficDefinition = undefined;
+      expect(uquery.getQueryDefinition()).toBe(querydefinition)
+      uquery = renewUquery();
+      uquery.trafficDefinition = querydefinition;
+      uquery.queryDefinition = undefined;
+      uquery.chartDefinition = undefined;
+      expect(uquery.getQueryDefinition()).toBe(querydefinition)
+      done();
```

Comment:
Please remove the function parameter *done* and the line above. This doesn't execute asynchronously.